src/foundry/client/core/issues.d.mts

```diff
@@ -0,0 +1,196 @@
+import type { DocumentConstructor, DocumentType } from "../../../types/helperTypes.d.mts";
+import type { DataModelValidationError } from "../../common/data/validation-failure.d.mts";
+
+declare global {
+  /**
+   * An object structure of document types at the top level, with a count of different sub-types for that document type.
+   */
+  interface ModuleSubTypeCounts extends Record<DocumentType, Record<string, number>> {}
+
+  /**
+   * A class responsible for tracking issues in the current world.
+   */
+  class ClientIssues {
+    /**
+     * Keep track of valid Documents in the world that are using module-provided sub-types.
+     * @remark the keys are module ids.
+     */
+    #moduleTypeMap: Map<string, ModuleSubTypeCounts>;
+
+    /**
+     * Keep track of document validation failures.
+     */
+    #documentValidationFailures: Record<
+      DocumentType,
+      Record<string, { name: string; error: DataModelValidationError }>
+    >;
+
+    /**
+     * Keep track of any usability issues related to browser or technology versions.
+     */
+    #usabilityIssues: Record<string, ClientIssues.UsabilityIssue>;
+
+    /** The minimum supported resolution. */
+    static #MIN_RESOLUTION: { WIDTH: 1024; HEIGHT: 700 };
+
+    /**
+     * The minimum supported client versions.
+     */
+    static #BROWSER_TESTS: {
+      /**
+       * @defaultValue
+       * ```typescript
+       * {
+       *   minimum: 24;
+       *   match: RegExp;
+       *   message: "ERROR.ElectronVersion";
+       * }
+       * ```
+       */
+      Electron: ClientIssues.BROWSER_TEST;
+      /**
+       * @defaultValue
+       * ```
+       * {
+       *   minimum: 92;
+       *   match: RegExp;
+       *   message: "ERROR.BrowserVersion";
+       * }
+       * ```
+       */
+      Chromium: ClientIssues.BROWSER_TEST;
+      /**
+       * @defaultValue
+       * ```
+       * {
+       *   minimum: 90;
+       *   match: RegExp;
+       *   message: "ERROR.BrowserVersion";
+       * }
+       * ```
+       */
+      Firefox: ClientIssues.BROWSER_TEST;
+      /**
+       * @defaultValue
+       * ```
+       * {
+       *   minimum: 15.4;
+       *   match: RegExp;
+       *   message: "ERROR.BrowserVersion";
+       * }
+       * ```
+       */
+      Safari: ClientIssues.BROWSER_TEST;
+    };
+
+    /**
+     * Add a Document to the count of module-provided sub-types.
+     * @param documentName      - The Document name.
+     * @param subType           - The Document's sub-type.
+     * @param options.decrement - Decrement the count rather than incrementing it. default: `false`
```

Comment:
To properly document the sub-properties, you actually do them in line with the block, e.g.
```ts
options?: {
/** 
  * Decrement the count rather than incrementing it.
  * @defaultValue `false`
  */
  decrement?: boolean
}
```

Note also the use of `defaultValue` here.

P.S. you can still leave out the `options` annotation since there's no comment.